src/vm/hints/hint_utils.rs

```diff
@@ -739,6 +739,76 @@ pub fn is_positive(
     }
 }
 
+//Implements hint:
+// %{
+//     from starkware.cairo.common.math_utils import assert_integer
+//     assert ids.MAX_HIGH < 2**128 and ids.MAX_LOW < 2**128
+//     assert PRIME - 1 == ids.MAX_HIGH * 2**128 + ids.MAX_LOW
+//     assert_integer(ids.value)
+//     ids.low = ids.value & ((1 << 128) - 1)
+//     ids.high = ids.value >> 128
+// %}
+pub fn split_felt(
+    vm: &mut VirtualMachine,
+    ids: HashMap<String, BigInt>,
+) -> Result<(), VirtualMachineError> {
+    //Check that ids contains the reference id for the variables used by the hint
+    let (high_ref, low_ref, value_ref) = if let (Some(high_ref), Some(low_ref), Some(value_ref)) = (
+        ids.get(&String::from("high")),
+        ids.get(&String::from("low")),
+        ids.get(&String::from("value")),
+    ) {
+        (high_ref, low_ref, value_ref)
+    } else {
+        return Err(VirtualMachineError::IncorrectIds(
+            vec![
+                String::from("high"),
+                String::from("low"),
+                String::from("value"),
+            ],
+            ids.into_keys().collect(),
+        ));
+    };
+
+    // Get the addresses of the variables used in the hints
+    let (high_addr, low_addr, value_addr) =
+        if let (Some(high_addr), Some(low_addr), Some(value_addr)) = (
+            get_address_from_reference(high_ref, &vm.references, &vm.run_context, vm),
+            get_address_from_reference(low_ref, &vm.references, &vm.run_context, vm),
+            get_address_from_reference(value_ref, &vm.references, &vm.run_context, vm),
+        ) {
+            (high_addr, low_addr, value_addr)
+        } else {
+            return Err(VirtualMachineError::FailedToGetIds);
+        };
+
+    //Check that the 'value' variable is in memory
+    match vm.memory.get(&value_addr) {
+        Ok(Some(maybe_rel_value)) => {
+            //Main logic
+            //assert_integer(ids.value)
+            let value = if let MaybeRelocatable::Int(ref value) = maybe_rel_value {
+                value
+            } else {
+                return Err(VirtualMachineError::ExpectedInteger(value_addr.clone()));
```

Comment:
According to [codecov](https://app.codecov.io/gh/lambdaclass/cleopatra_cairo/blob/split-felt_hint-implementation/src/vm/hints/hint_utils.rs) this case is not covered and it seems to be part of the hint logic, do you think you could add a test for it?

Reply:
Thanks, Juan, I forgot about this one,
Sure, I will add one

Reply:
[Done](https://github.com/lambdaclass/cleopatra_cairo/pull/251/commits/ffa60cd4f80ff824cfc0a8088c700a2bf7fb0119)